pkg/apim/APIMUtility.go

```diff
@@ -51,3 +66,21 @@ func (apim *Bundle) IsPayloadJSON() bool {
 	var js json.RawMessage
 	return json.Unmarshal([]byte(apim.Payload), &js) == nil
 }
+
+func (odataFilters *OdataParameters) InitOdataQuery() (string, error) {
+
+	v := reflect.ValueOf(odataFilters).Elem()
+	typeOfS := v.Type()
+	urlParam := url.Values{}
+	for i := 0; i < v.NumField(); i++ {
+		structVal := fmt.Sprintf("%v", v.Field(i).Interface())
+		structVal = strings.Replace(structVal, "[", "", 1)
+		structVal = strings.Replace(structVal, "]", "", 1)
```

Comment:
This looks like very complicated stuff. First: What is happening in this function? Second: How can we make the code more understandable?

Reply:
removed